openedx_events/data.py

```diff
@@ -47,36 +47,36 @@ class EventsMetadata:
             lexical ordering of strings (e.g. '0.9.0' vs. '0.10.0').
     """
 
-    id = attr.ib(type=UUID, init=False)
+    id = attr.ib(
+        type=UUID, default=None,
+        converter=attr.converters.default_if_none(attr.Factory(lambda: uuid1())),  # pylint: disable=unnecessary-lambda
```

Comment:
**Note to Reviewer**: 
1. The pylint disable is because it was simpler to mock and test with a lambda.
2. The following `default_if_none` converters were all based on code that was in `__attrs_post_init__`, but was removed in this PR.

Please mark resolved once read by main reviewer.